.github/workflows/plugin-CI.yml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js 16.x
+      uses: actions/setup-node@v2
+      with:
+        node-version: 16.x
+    - name: Install Yarn
+      run: npm install -g yarn
+      
+    - name: Install dependencies
+      run: yarn install --frozen-lockfile
+
+    - name: Run tests
+      run: |
+        yarn test |& tee /tmp/test.log
+        if grep -q "console.warn" /tmp/test.log; then
+          echo "::warning::Tests produced warnings!"
+        fi
+        if grep -q "Test Suites:.* failed" /tmp/test.log; then
+          exit 1
+        fi
```

Comment:
i don't really see the point of this. `yarn test` should return an exit code > 0 if some test are failing.
and if you need the logs, you can just go in github action page